private/pkg/git/git.go

```diff
@@ -376,16 +383,45 @@ func OpenRepositoryWithDefaultBranch(name string) OpenRepositoryOption {
 	}
 }
 
-// referenceType is the type of references that can be passed as starting points when traversing a
-// git tree.
-type referenceType string
+type forEachBranchOpts struct {
+	remote string
+}
+
+type headCommitOpts struct {
+	branch string
+	remote string
+}
+
+type forEachCommitWithBranchStartPointOpts struct {
+	remote string
+}
 
 // reference is a single git reference used in ForEachCommit to declare an starting commit.
-type reference struct {
-	refName string
-	refType referenceType
+type reference interface {
```

Comment:
This should go in `reference.go`.

Reply:
How do you differentiate from other unexported func opts? The reference is also related just to func opts.

Reply:
Ah that's fair, you're right.